app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -365,18 +365,30 @@ def confirmation_for_add_phone?
   end
 
   def presenter_for_two_factor_authentication_method
-    type = DELIVERY_METHOD_MAP[two_factor_authentication_method.to_sym]
-
-    return unless type
-
-    data = send("#{type}_view_data".to_sym)
-
-    TwoFactorAuthCode.const_get("#{type}_delivery_presenter".classify).new(
-      data: data,
-      view: view_context,
-      service_provider: current_sp,
-      remember_device_default: remember_device_default,
-    )
+    case two_factor_authentication_method.to_sym
+    when :authenticator
+      TwoFactorAuthCode::AuthenticatorDeliveryPresenter.new(
+        data: authenticator_view_data,
+        view: view_context,
+        service_provider: current_sp,
+        remember_device_default: remember_device_default,
+      )
+    when :sms, :voice
+      TwoFactorAuthCode::PhoneDeliveryPresenter.new(
+        data: phone_view_data,
+        view: view_context,
+        service_provider: current_sp,
+        remember_device_default: remember_device_default,
+      )
+    when :piv_cac
+      # NOTE: this does not match, is this branch ever called?
```

Comment:
maybe not?

https://github.com/18F/identity-idp/blob/64b454ad42eff13da6e1ac45a2fc9a76d0666600/app/controllers/two_factor_authentication/piv_cac_verification_controller.rb#L95-L96 redefines the `presenter_for_two_factor_authentication_method` method and creates an instance directly

Reply:
ok yup we're on the same page. gonna yank it